src/core/caniuse.js

```diff
@@ -0,0 +1,172 @@
+  (otherwise `DEFAULT_BROWSERS` will be used).
+*/
+
+import { fetch as ghFetch } from "core/github";
+import { semverCompare } from "core/utils";
+import { pub } from "core/pubsubhub";
+import "deps/hyperhtml";
+import caniuseCss from "deps/text!core/css/caniuse.css";
+
+// Opportunistically insert the style into the head to reduce FOUC.
+const codeStyle = document.createElement("style");
+codeStyle.textContent = caniuseCss;
+document.head.appendChild(codeStyle);
+
+export const name = "core/caniuse";
+
+export async function run(conf) {
+  if (conf.caniuse) {
+    const el = document.querySelector(".head dl");
+    canIUse(conf.caniuse, el, conf.caniuseBrowsers);
+  }
+  for (const el of document.querySelectorAll("section[data-caniuse]")) {
+    canIUse(el.dataset.caniuse, el.firstChild, conf.caniuseBrowsers);
+  }
+}
+
+const DEFAULT_BROWSERS = ["chrome", "firefox", "ie", "edge"];
+const BROWSERS = { // browser name dictionary
+  chrome: "Chrome",
+  firefox: "Firefox",
+  ie: "IE",
+  edge: "Edge",
+  android: "Android",
+  safari: "Safari",
+  opera: "Opera",
+  bb: "Blackberry",
+  and_uc: "UC (Android)",
+  and_ff: "Firefox (Android)",
+};
+
+/**
+ * main canIUse function
+ * @param  {String} key                         which api to look for
+ * @param  {Node} el                          add table after el
+ * @param  {Array:string} [browsers=DEFAULT_BROWSERS] list of browsers to show
+ */
+function canIUse(key, el, browsers = DEFAULT_BROWSERS) {
+  const url = `https://raw.githubusercontent.com/Fyrd/caniuse/master/features-json/${key}.json`;
+
+  // use data from localStorage data if valid and render
+  const cached = localStorage.getItem(`caniuse-${key}`);
+  if (cached) {
+    const stats = JSON.parse(cached);
+    const CACHE_DURATION = 24 * 60 * 60 * 1000; // in ms
+    if (new Date() - new Date(stats.$cacheTime) < CACHE_DURATION) {
+      return showData(key, stats, el, browsers);
+    }
+  }
+  // otherwise fetch new data and render
+  const placeholder = createPlaceholder(key, el);
+  const handleResponse = ({ stats }) => showData(key, stats, el, browsers);
+  const handleError = err => showError(err, placeholder);
+  ghFetch(url)
+    .then(handleResponse)
+    .catch(handleError);
+}
+
+/**
+ * creates a placeholder while the API fetches results
+ * @param  {String} key    API name
+ * @param  {Node} parent where to create placeholder
+ * @return {Node}        inserted placeholder
+ */
+function createPlaceholder(key, parent) {
+  const canIUseId = `caniuse-${key}`;
+  const placeholder = hyperHTML`
+    <div class="caniuse" id="${canIUseId}">
+      <dt class="caniuse-title">
+        Can I Use this API? (${key})
+      </dt>
+      <dd>fetching data from caniuse.com... </dl>
+    </div>`;
+  return parent.parentNode.insertBefore(placeholder, parent.nextSibling);
+}
+
+function showError(err, placeholder) {
+  placeholder.style.height = "40px";
+  placeholder.querySelector("dd").innerText = `Error: ${err.message}`;
```

Comment:
We need to consider what the error might be... often, the reader can't do much about these errors - so maybe the best recovery would be to just add a link to the canIuse.com page. We can then route the actual error to `console.error()`.